src/migrations/migrate.rs

```diff
@@ -147,6 +149,13 @@ pub async fn migrate(cli: &mut Connection, _options: &Options,
         }
         return Ok(());
     }
+
+    // Disable idle transaction timeout while migrations get applied.
+    let old_timeout = cli.query_row::<Duration, _>(
+        "SELECT cfg::Config.session_idle_transaction_timeout", &()).await?;
+    cli.execute("CONFIGURE SESSION SET session_idle_transaction_timeout \
```

Comment:
Just set it inside the transaction? Then you don't need to save the old value.

Reply:
This is what I do in our server-side dump implementation https://github.com/edgedb/edgedb/blob/3402eb6f2b4877c1273b9bdf7e98b99442b979d5/edb/server/protocol/binary.pyx#L2396-L2409

Reply:
This only works if you `ROLLBACK` at the end.  We don't have `CONFIGURE TRANSACTION` yet, unfortunately.

Reply:
ah, true